var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -88,6 +96,27 @@ class QuantumEspresso(Package):
     # folder QE expects as a link, we issue a conflict here.
     conflicts('+elpa', when='@:5.4.0')
 
+    # The first version of Q-E to feature integrated EPW is 6.0.0,
+    # as per http://epw.org.uk/Main/DownloadAndInstall .
+    # Complain if trying to install a version older than this.
+    conflicts('+epw', when='@:5',
+            msg='EPW only available from version 6.0.0 and on')
+
+    # Below goes some constraints as shown in the link above.
+    # Constraints may be relaxed as successful reports
+    # of different compiler+mpi combinations arrive
+
+    # KISS for now and only build the mpi version
+    conflicts('+epw', when='~mpi', msg='EPW needs MPI')
+
+    # EPW doesn't gets along well with OpenMPI 2.x.x
+    conflicts('+epw', when='^openmpi@2.0.0:2.999.999',
+            msg='OpenMPI version incompatible with EPW')
+
+    # EPW also doesn't gets along well with PGI 17.x + OpenMPI 1.10.7
```

Comment:
I think the `when` argument has all the info in this comment so it isn't needed (if you have the error it reports though that would be useful in a comment).

Reply:
Those are actually two different situations, as per [EPW install page](http://epw.org.uk/Main/DownloadAndInstall):
- Any openmpi version of the 2.x.x series
- Openmpi 1.10.7 compiled by any PGI release of the 17.x series

So IMO those two different `conflict(...)` are needed.
(It may be over-cautious, but I'd rather play it safe for now)

Reply:
@fcannini I wasn't complaining about the conflicts, just the comments: in my opinion, they didn't add any information that wasn't already part of the actual `conflict` statement.

Reply:
Aaah I see, sorry if I misread you. Sometimes I prose in the comments more than needed. :)